### PR TITLE
feat(workspace/app_server_group): add resource to manage server group

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:
Add new resource to manage a workspace APP server group.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource.
2. add corresponding document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
./scripts/coverage.sh -o workspace -f TestAccResourceAppServerGroup_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccResourceAppServerGroup_basic -timeout 360m -parallel 10
=== RUN   TestAccResourceAppServerGroup_basic
=== PAUSE TestAccResourceAppServerGroup_basic
=== CONT  TestAccResourceAppServerGroup_basic
--- PASS: TestAccResourceAppServerGroup_basic (50.87s)
PASS
coverage: 6.8% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 50.927s coverage: 6.8% of statements in ./huaweicloud/services/workspace
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
   image

     
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/resources/workspace_app_server_group.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  + **gold**: The market image.
+  + **public**: The public image.
+  + **private**: The private image.
+  + **shared**: The shared image.
+  + **other**
+
+* `image_product_id` - (Optional, String) Specifies the image product ID of the server group.  
+  This parameter is required wnen the `image_type` parameter is set to **gold**.
+
+* `is_vdi` - (Optional, Bool, ForceNew) Specifies the session mode of the server group. Defaults to `false`.
```
Comment (ruwenqiang123) [inline docs/resources/workspace_app_server_group.md]:
```suggestion
* `is_vdi` - (Optional, Bool, ForceNew) Specifies the session mode of the server group. Defaults to **false**.
```

File: docs/resources/workspace_app_server_group.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  If omitted, the AZ randomly assigned by the system is used.  
+  Changing this creates a new resource.
+
+* `description` - (Optional, String) Specifies the description of the server group.
+
+* `tags` - (Optional, Map, ForceNew) Specifies the key/value pairs to associate with the server group.
+  Supports up to 20 tags.  
+  Changing this creates a new resource.
+  
+* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID of the server group.
```
Comment (ruwenqiang123) [inline docs/resources/workspace_app_server_group.md]:
```suggestion
* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID to which the server group belongs.  
```

File: docs/resources/workspace_app_server_group.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  + **CPU**
+  
+* `extra_session_size` - (Optional, Int, ForceNew) Specifies the number of additional sessions for a single server.
+  Changing this creates a new resource.  
+  This parameter is available only wnen the `is_vdi` parameter is set to **false**.  
+  The `extra_session_size` must be used together with `extra_session_type`.  
+  The upper limit of the number of additional sessions for a single server is `10` times the number of vCPUs in the server
+  specification minus the default number of sessions in the package.
+
+* `primary_server_group_id` - (Optional, String, ForceNew) Specifies the ID of the primary server group ID.
```
Comment (ruwenqiang123) [inline docs/resources/workspace_app_server_group.md]:
```suggestion
* `primary_server_group_id` - (Optional, String, ForceNew) Specifies the ID of the primary server group.
```

**Comment by dengyali0125** [comment]
LGTM

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccResourceAppServerGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccResourceAppServerGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceAppServerGroup_basic
=== PAUSE TestAccResourceAppServerGroup_basic
=== CONT  TestAccResourceAppServerGroup_basic
--- PASS: TestAccResourceAppServerGroup_basic (59.45s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 59.527s

**Pull request merged**